cmd/deploy/endpoints.go

```diff
@@ -180,23 +195,33 @@ func (eg *EndpointGetter) getEndpoints(ctx context.Context, opts *EndpointsOptio
 
 	sanitizedName := format.ResourceK8sMetaString(opts.Name)
 	labelSelector := fmt.Sprintf("%s=%s", model.DeployedByLabel, sanitizedName)
-	iClient, err := eg.K8sClientProvider.GetIngressClient()
-	if err != nil {
-		return nil, err
-	}
-	eps, err := iClient.GetEndpointsBySelector(ctx, opts.Namespace, labelSelector)
-	if err != nil {
-		return nil, err
-	}
 
-	externalEps, err := eg.endpointControl.List(ctx, opts.Namespace, labelSelector)
-	if err != nil {
-		return nil, err
-	}
+	var eps []string
+	var err error
+	if okteto.Context().IsOkteto {
```

Comment:
I think this should be an interface. This way we can encapsulate it into a single function making it more readable and the resulting test would be more readable. There is no need if I'm on a Vanilla cluster to have access to `endpointControl` and `externalResourceControl`WDYT?

Reply:
e51c9a1cd38266d5e515e36628b2c415714d6e85